### PR TITLE
Complete transition to Artifacts stdlib

### PR DESCRIPTION
**Pull request by staticfloat**
This bumps the `Pkg` version to one that strips out the `Pkg.Artifacts` and `Pkg.BinaryPlatforms` modules.

### CODE REVIEW

File: stdlib/Pkg.version [hunk 1]
```diff
@@ -1,2 +1,2 @@
 PKG_BRANCH = master
-PKG_SHA1 = 6a235eb813be335b54c97c5c7d631bdbd1059115
+PKG_SHA1 = c2c99807b157d6b9e23d09fa597800ec0e715cee
```
Comment (fredrikekre) [inline stdlib/Pkg.version:2]: Hashes?

Comment (staticfloat) [reply stdlib/Pkg.version:2]: I'm not sure what you're asking.  :P

Comment (staticfloat) [reply stdlib/Pkg.version:2]: AH, you're asking for the checksums to be added.  Yes, good call.

**Comment by staticfloat** [comment]
@nanosoldier `runtests(ALL, vs = ":master")`

**Comment by staticfloat** [comment]
I think Mosè accidentally tracked down the issue that was causing problems on PkgEval.

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/37592#issuecomment-694549685) has completed - possible new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/4be025709237d90f5e98ce854b94681838cd1980/pkgeval/by_hash/cdb2a72_vs_73413fc/report.md). cc @maleadt

**Pull request merged**